### PR TITLE
:sparkles: decomm and comm longevity test

### PR DESCRIPTION
**Pull request by lsrinivas-pure**
**What this PR does / why we need it**:
Addint node decommssion and rejoin as part of longevity

**Which issue(s) this PR fixes** (optional)
Closes # PT-4703
**Special notes for your reviewer**:

Decomm node:

STEP: Decommission a random node
STEP: decommission node torpedo-long-fa-sample-test-36-4
INFO[2022-05-17 17:43:27] testAndSetEndpoint failed for 10.233.17.201: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 10.233.17.201:9020: i/o timeout"
INFO[2022-05-17 17:43:27] Getting new driver.
INFO[2022-05-17 17:43:29] Using 10.13.10.193:9020 as endpoint for portworx volume driver
DEBU[2022-05-17 17:43:29] Inspecting node [torpedo-long-fa-sample-test-36-4] with volume driver node id [557773e0-bcb0-4c43-b9ca-8f0488a0371a]
INFO[2022-05-17 17:44:00] testAndSetEndpoint failed for 10.233.17.201: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 10.233.17.201:9020: i/o timeout"
INFO[2022-05-17 17:44:00] Getting new driver.
INFO[2022-05-17 17:44:02] Using 10.13.10.193:9020 as endpoint for portworx volume driver
INFO[2022-05-17 17:44:02] removing node 557773e0-bcb0-4c43-b9ca-8f0488a0371a
INFO[2022-05-17 17:44:11]  557773e0-bcb0-4c43-b9ca-8f0488a0371a: Node remove is pending
INFO[2022-05-17 17:45:02] testAndSetEndpoint failed for 10.233.17.201: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 10.233.17.201:9020: i/o timeout"
INFO[2022-05-17 17:45:02] Getting new driver.
INFO[2022-05-17 17:45:03] Waiting for lock for trigger [nodeRejoin]
INFO[2022-05-17 17:45:04] Using 10.13.8.182:9020 as endpoint for portworx volume driver
INFO[2022-05-17 17:45:06] Successfully removed node 557773e0-bcb0-4c43-b9ca-8f0488a0371a

Rejoin node:

STEP: Rejoin the node
STEP: Rejoin node torpedo-long-fa-sample-test-36-4
DEBU[2022-05-17 17:45:43] Finding the debug pod to run command on node torpedo-long-fa-sample-test-36-4
DEBU[2022-05-17 17:45:44] Running command on pod debug-265tc [[nsenter --mount=/hostproc/1/ns/mnt /bin/bash -c which pxctl]]
DEBU[2022-05-17 17:45:46] Finding the debug pod to run command on node torpedo-long-fa-sample-test-36-4
DEBU[2022-05-17 17:45:47] Running command on pod debug-265tc [[nsenter --mount=/hostproc/1/ns/mnt /bin/bash -c sudo /usr/local/bin/pxctl sv node-wipe --all]]
INFO[2022-05-17 17:45:56] Node wipe is successfull
INFO[2022-05-17 17:45:57] Remove service label is successfull
INFO[2022-05-17 17:45:57] Remove enable label is successfull
INFO[2022-05-17 17:45:58] Restart Driver is successfull
INFO[2022-05-17 17:45:58] Uncordon is successfull
INFO[2022-05-17 17:46:43] Waiting for lock for trigger [nodeDecomm]
INFO[2022-05-17 17:47:30] Parsed node [torpedo-long-fa-sample-test-36-0] as Type: Worker, Zone: , Region
INFO[2022-05-17 17:47:30] Parsed node [torpedo-long-fa-sample-test-36-1] as Type: Worker, Zone: , Region
INFO[2022-05-17 17:47:31] Parsed node [torpedo-long-fa-sample-test-36-2] as Type: Worker, Zone: , Region
INFO[2022-05-17 17:47:31] Parsed node [torpedo-long-fa-sample-test-36-3] as Type: Master, Zone: , Region
INFO[2022-05-17 17:47:32] Parsed node [torpedo-long-fa-sample-test-36-4] as Type: Worker, Zone: , Region
INFO[2022-05-17 17:47:32] Parsed node [torpedo-long-fa-sample-test-36-5] as Type: Worker, Zone: , Region
INFO[2022-05-17 17:47:53] testAndSetEndpoint failed for 10.233.17.201: rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 10.233.17.201:9020: i/o timeout"
INFO[2022-05-17 17:47:53] Getting new driver.
INFO[2022-05-17 17:47:55] Using 10.13.8.190:9020 as endpoint for portworx volume driver
INFO[2022-05-17 17:47:57] Latest Node: torpedo-long-fa-sample-test-36-1
INFO[2022-05-17 17:47:57] Latest Node: torpedo-long-fa-sample-test-36-5
INFO[2022-05-17 17:47:57] Latest Node: torpedo-long-fa-sample-test-36-0
INFO[2022-05-17 17:47:57] Latest Node: torpedo-long-fa-sample-test-36-2
INFO[2022-05-17 17:47:57] Latest Node: torpedo-long-fa-sample-test-36-4
INFO[2022-05-17 17:47:57] node torpedo-long-fa-sample-test-36-4 rejoin is successful

### CODE REVIEW

File: drivers/volume/portworx/portworx.go [hunk 1]
```diff
@@ -2461,29 +2467,37 @@ func (d *portworx) DecommissionNode(n *node.Node) error {
...
 
-	if err := d.StopDriver([]node.Node{*n}, false, nil); err != nil {
+	if err := d.EnterMaintenance(*n); err != nil {
 		return &ErrFailedToDecommissionNode{
 			Node:  n.Name,
 			Cause: fmt.Sprintf("Failed to stop driver on node: %v. Err: %v", n.Name, err),
 		}
 	}
 
+	time.Sleep(10 * time.Second)
```
Comment (jainsmit) [inline drivers/volume/portworx/portworx.go:2477]: why code is sleep for 10 second here? Can we replace sleep with wait method to make sure node went into maintenance mode...

File: drivers/volume/portworx/portworx.go [hunk 2]
```diff
@@ -2461,29 +2467,37 @@ func (d *portworx) DecommissionNode(n *node.Node) error {
...
+		if !strings.Contains(err.Error(), "Node remove is pending") {
+			return &ErrFailedToDecommissionNode{
+				Node:  n.Name,
+				Cause: err.Error(),
+			}
 		}
 	}
 
+	logrus.Infof(" %v: Node remove is pending. Waiting for it to complete", nodeResp.Node.Id)
+	time.Sleep(30 * time.Second)
```
Comment (jainsmit) [inline drivers/volume/portworx/portworx.go:2499]: Sleep need to be replaced with wait and check method ...

File: drivers/volume/portworx/portworx.go (hunk 1 above)
Comment (lsrinivas-pure) [reply drivers/volume/portworx/portworx.go:2477]: sleep not required here.Fixed it

File: tests/testTriggers.go [hunk 3]
```diff
@@ -3619,6 +3627,168 @@ func waitForFsTrimStatus(event *EventRecord, attachedNode, volumeID string) stri
...
+		},
+		Start:   time.Now().Format(time.RFC1123),
+		Outcome: []error{},
+	}
+
+	defer func() {
+		event.End = time.Now().Format(time.RFC1123)
+		*recordChan <- event
+	}()
+
```
[the remark this answers is not in the document]
Comment (lsrinivas-pure) [reply tests/testTriggers.go:3646]: these defer calls should be called insid the method only as these a linked to test method

File: drivers/volume/portworx/portworx.go (hunk 2 above)
Comment (lsrinivas-pure) [reply drivers/volume/portworx/portworx.go]: fixed it

**Review by jainsmit — APPROVED** [review]

**Pull request merged**